src/main/java/org/jboss/modules/ModuleIdentifier.java

```diff
@@ -40,9 +42,13 @@ public final class ModuleIdentifier implements Serializable {
 
     private static final String DEFAULT_SLOT = "main";
 
+    private static Map<String, ModuleIdentifier> identifierMap = new ConcurrentHashMap<>();
+
     private final String name;
     private final String slot;
 
+    private String escapedString;
+
```

Comment:
Idiomatically, we usually would call this field `toString` in order to match up with the method from which it is used. And, it would need to be `transient`.